### DISCUSSION TITLE
Dirty read, Non-repeatable read, Read skew의 차이점

### DISCUSSION THREAD

**Discussion by yoondgu** [woowacourse-study/2023-db-db-deep#8] [Q&A]
Dirty read, Non-repeatable read, Read skew의 차이점을 정리해봅시다

**Comment by yeonkkk** [comment]
혼자 한줄 정리  

- 더티 리드(Dirty Read)는 커밋되지 않은 변화를 읽어서 문제가 발생하는 현상입니다.
- Non-Repeatable Read는 같은 데이터를 여러번 읽었을 때 값이 달라지는 현상을 말합니다.
-  Read Skew는 다수의 트랜잭션이 동시에 데이터를 읽고 쓰는 환경에서 발생하는 데이터 일관성 문제입니다.  


- 비슷한데 약간 다른 두 현상: non-repeatable read vs read skew 
  -  non-repeatable read 는 하나의 트랜잭션 내에서 같은 데이터를 여러번 읽었는데 결과가 다른 현상
  - read skew는 동시에 같은 데이터를 읽는 사이에 쓰기 작업이 발생하여 같은 읽기 작업을 했음에도 결과가 상이하게 나오는 현상

**Reply by yeonkkk** [reply]
ref. 
지선생님  
https://jiwondev.tistory.com/163

**Comment by yoondgu** [comment]
- Dirty read : 아직 커밋되지 않은 데이터를 다른 트랜잭션에서 읽을 수 있는 문제 (데이터 정합성이 깨질 수 있음)
- Non-repeatable read : 같은 데이터에 대해 여러 번 조회할 때, 다른 값을 읽어 일관성을 보장하지 못하는 문제
- Read skew : 서로 연관이 있는 두 개의 데이터에 대해 조회할 때, 조회하는 사이 쓰기 작업이 발생하여 연관된 데이터 간의 일관성을 보장하지 못하는 문제
